R/completion.r

```diff
@@ -45,20 +45,32 @@ completions <- function(code, cursor_pos = nchar(code), fixup = TRUE) {
 fixup_comps <- function(comps) {
     # TODO: only do this if we are not in a string or so
     re_trail <- '=|::'
-    re_lead <- '[\\w\\d._]+(?:\\$|@|:::?)'  # TODO: allow foo$`_bar`$baz<tab>
```

Comment:
R itself doesn't do auto-completion for this case, so not sure there's anything to be done on the IRkernel side:

```r
foo = list(`_bar` = list(bazaar = 1))
foo$`_bar`$baz<tab> # nothing
```